TabSanity/ArrowKeyFilter.cs

```diff
@@ -119,6 +120,15 @@ public override int Exec(ref Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt,
 			return NextTarget.Exec(ref pguidCmdGroup, nCmdID, nCmdexecopt, pvaIn, pvaOut);
 		}
 
+		ITextSnapshotLine FindNextLine()
+		{
+			var current = TextView.TextSnapshot.GetLineFromPosition(Caret.Position.BufferPosition.Position);
```

Comment:
Please cache `TextView.TextSnapshot` and `Caret.Position.BufferPosition` at the top of this function for reuse.

```cs
var snapshot = TextView.TextSnapshot;
var caretBufferPosition = Caret.Position.BufferPosition;
```